drivers/at/at.c

```diff
@@ -88,18 +101,49 @@ static void _isrpipe_write_one_wrapper(void *_dev, uint8_t data)
 #endif
 }
 
-int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize)
+int at_dev_init(at_dev_t *dev, at_dev_init_t const *init)
 {
-    dev->uart = uart;
+    dev->uart = init->uart;
+    dev->rp_buf = init->rp_buf;
+    dev->rp_buf_size = init->rp_buf_size;
 
 #if IS_USED(MODULE_AT_URC_ISR)
     dev->awaiting_response = false;
     dev->event.handler = _event_process_urc;
 #endif
 
-    isrpipe_init(&dev->isrpipe, (uint8_t *)buf, bufsize);
+    isrpipe_init(&dev->isrpipe, (uint8_t *)init->rx_buf, init->rx_buf_size);
 
-    return uart_init(uart, baudrate, _isrpipe_write_one_wrapper, dev);
+    return uart_init(init->uart, init->baudrate, _isrpipe_write_one_wrapper, dev);
+}
+
+int at_parse_resp(at_dev_t *dev, char const *resp)
+{
+    if (*resp == '\0') {
+        return 1;
+    }
+    if (starts_with(resp, CONFIG_AT_RECV_OK)) {
+        dev->rp_buf[0] = '\0';
+        return 0;
+    }
+    if (starts_with(resp, CONFIG_AT_RECV_ERROR)) {
+        return -1;
+    }
+    // A specific command may return either CME or CMS, we need not differentiate
+    if (!starts_with(resp, "+CME ERROR: ") &&
+        !starts_with(resp, "+CMS ERROR: ")) {
+        // neither `OK` nor error, must be a response or URC
+        return 1;
+    }
+    
+    resp += strlen("+CMx ERROR: ");    
+    size_t resp_len = strlen(resp);
+    if (resp_len + 1 > dev->rp_buf_size) {
```

Comment:
```suggestion
    size_t resp_len = strlen(resp) + 1;
    if (resp_len > dev->rp_buf_size) {
```

is a bit nicer to read

Reply:
It means response **string** length and I want to keep it consistent across the whole file, otherwise it might get confusing quickly.